router/core/src/main/scala/io/buoyant/rinet.scala

```diff
@@ -0,0 +1,22 @@
+package io.buoyant
+
+import com.twitter.finagle._
+import com.twitter.util.Activity
+
+/**
+ * An alternate form of the inet namer that expects port before host.
+ *
+ *   /$/io.buoyant.rinet/<port>/<host>
+ *
+ * is equivalent to
+ *
+ *   /$/inet/<host>/<port>
+ */
+class rinet extends Namer {
+  override def lookup(path: Path): Activity[NameTree[Name]] = path.take(2) match {
+    case Path.Utf8(port, host) =>
+      Activity.value(NameTree.Leaf(Resolver.eval(s"inet!$host:$port")))
+    case _ =>
+      Activity.value(NameTree.Neg)
+  }
+}
```

Comment:
Alternatively, consider just making this a rewriting namer that rewrites to `Path.Utf8("$", "inet", host, port)` -- the idea being that if the implementation of /$/inet changes, we'll get it for free.